### ISSUE TITLE
Field this.serverless.variables.options is undefined from 2.29.0

### ISSUE THREAD

**Issue by PatrykMilewski** [serverless/serverless#9099]
Starting from Serverless 2.29.0 I've noticed, that some plugins have issues with input `serverless` object.

Until 2.28.0, it was possible to get options passed to Serverless CLI by using `this.serverless.variables.options`. However starting from 2.29.0, options field is always undefined, which causes for example `serverless-plugin-resource-tagging` to break (it's trying to read options using this field).

I did also noticed, that this issue occurs only when invoking Serverless using Yarn/NPM script, which might somehow break passing options to CLI?

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: Api

plugins:
  - serverless-plugin-resource-tagging

provider:
  name: aws
  stage: ${opt:stage}
  runtime: nodejs12.x
```

</details>

<details>
<summary><b><code>yarn run package --stage anystagename --region eu-west-1</code> output</b></summary>

```
yarn run package --stage staging --region eu-west-1
yarn run v1.22.10
warning package.json: No license field
$ sls package --stage staging --region eu-west-1
Serverless: To ensure safe major version upgrades ensure "frameworkVersion" setting in service configuration (recommended setup: "frameworkVersion: ^2.29.0")

Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command config:tabcompletion
Serverless: Load command config:tabcompletion:install
Serverless: Load command config:tabcompletion:uninstall
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Load command output
Serverless: Load command output:get
Serverless: Load command output:list
Serverless: Load command param
Serverless: Load command param:get
Serverless: Load command param:list
Serverless: Load command studio
Serverless: Skipping variables resolution with old resolver (new resolver reported no more variables to resolve)
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Invoke aws:package:finalize
 
 Type Error ----------------------------------------------
 
  TypeError: Cannot read property 'stage' of undefined
      at ServerlessPlugin._addTagsToResource (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/serverless-plugin-resource-tagging/index.js:42:43)
      at /Users/patrykmilewski/WebstormProjects/MASKED/backend/api/node_modules/serverless/lib/classes/PluginManager.js:502:55
      at tryCatcher (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/util.js:16:23)
      at Object.gotValue (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/reduce.js:168:18)
      at Object.gotAccum (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/reduce.js:155:25)
      at Object.tryCatcher (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/promise.js:547:31)
      at Promise._settlePromise (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/promise.js:604:18)
      at Promise._settlePromise0 (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/promise.js:649:10)
      at Promise._settlePromises (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/promise.js:729:18)
      at _drainQueueStep (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/async.js:93:12)
      at _drainQueue (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/async.js:86:9)
      at Async._drainQueues (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/async.js:102:5)
      at Immediate.Async.drainQueues [as _onImmediate] (/Users/patrykmilewski/WebstormProjects/MASKED/backend/node_modules/bluebird/js/release/async.js:15:14)
      at processImmediate (internal/timers.js:461:21)
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          darwin
     Node Version:              12.19.1
     Framework Version:         2.29.0 (local)
     Plugin Version:            4.5.0
     SDK Version:               n/a
     Components Version:        3.7.3

```

</details>

<b>Installed version</b>

```
2.29.0
```

**Comment by thongle91** [comment]
Actually I believe it started from `2.28.1`, after this check

https://github.com/serverless/serverless/blob/v2.28.1/lib/Serverless.js#L235-L238

**Comment by pgrzesik** [comment]
@medikoo Would you be able to respond here? I know you have much better context of these changes

**Comment by medikoo** [comment]
It looks as another unfortunate case of plugin relying on Framework private parts, which are subject to change at any time.

Ideally if it's fixed on plugin side, and fixing this seems easy.
Instead of accessing `options` from `serverless.variables.options`, plugin should read `options` from object passed to plugin constructor.

@PatrykMilewski @thongle91 can you prepare a PR on plugin side?

**Comment by thongle91** [comment]
I have submitted a [PR](https://github.com/ilayanambi86/serverless-plugin-resource-tagging/pull/6/files) for that

**Comment by medikoo** [comment]
@thongle91 that's great to hear! If you approach problem with PR getting merged and release, let us know. We'll use best means to contact author of a plugin and have it published

**Comment by thongle91** [comment]
@medikoo the PR get merged and released in version 1.1.0.

**Comment by medikoo** [comment]
Closing as addressed on Plugin side. Still we've found that `variables.options` is necessary internally if some plugins for some reason use old variable resolver in their logic. Patch that ensures it's there will be posted shortly
